dev/src/util.ts

```diff
@@ -43,3 +48,49 @@ export function autoId(): string {
 export function requestTag(): string {
   return autoId().substr(0, 5);
 }
+
+/**
+ * Parses the arguments for the `getAll()` call supported by both the Firestore
+ * and Transaction class.
+ *
+ * @private
+ * @param validator The argument validator to use.
+ * @param documentRefsOrReadOptions An array of document references followed by
+ * an optional ReadOptions object.
+ */
+export function parseGetAllArguments(
+    validator: AnyDuringMigration,
+    documentRefsOrReadOptions: Array<DocumentReference|ReadOptions>):
+    {documents: DocumentReference[], fieldMask: FieldPath[]|null} {
+  let documents: DocumentReference[];
+  let readOptions: ReadOptions|undefined = undefined;
+
+  const usesVarags = !Array.isArray(documentRefsOrReadOptions[0]);
```

Comment:
This looks wrong to me. The function signature says that `documentRefsOrReadOptions` is of type `Array<DocumentReference|ReadOptions>`, so I would expect that the first element must be of type `DocumentReference|ReadOptions` and could never be an array.

Reply:
The non-TypeScript versions of this client supported a call to getAll with an array of DocumentReferences. I did not want to break this for no reason, especially since we have supported this through the prior TypeScript versions. There is already a comment on line 80, but I moved it up and made it much more explicit.